### PR TITLE
Vulnerabilities Corrected: Tar Slip and Path Traversal

### PR DESCRIPTION
**Pull request by ThewsyRum**
## Checklist

- [x] Read CONTRIBUTING.md, and accept the CLA by including the provided snippet. We will not accept PR without this.
- [ ] Run pre-commit hook.
- [ ] If you changed Rust code, run `cargo check`, `cargo clippy`, `cargo test`.

## PR Description

I, ThewsyRum, confirm that I have read and understood the terms of the CLA of Kyutai-labs, as outlined in the repository's CONTRIBUTING.md, and I agree to be bound by these terms.

Previously, calling extractall to unpack files from a tar archive didn't properly check the file paths, which could allow files to be extracted outside the intended directory.
`                tar.extractall(path=dist_tgz.parent)`

Input from a command line argument is passed directly to aiohttp.web.FileResponse as a path. This could create a Path Traversal vulnerability, enabling an attacker to read arbitrary files on the server.
`            return web.FileResponse(os.path.join(static_path, "index.html"))`

### CODE REVIEW

**Review by david6915 — APPROVED** [review]

File: moshi/moshi/server.py [hunk 1]
```diff
@@ -223,27 +236,53 @@ def main():
...
-    elif args.static != "none":
+    elif args.static.lower() != "none":
         # When set to the "none" string, we don't serve any static content.
         static_path = args.static
+
     if static_path is not None:
-        async def handle_root(_):
-            return web.FileResponse(os.path.join(static_path, "index.html"))
+        # Resolve the static path to an absolute path
+        static_path = Path(static_path).resolve()
```
Comment (adefossez) [inline moshi/moshi/server.py:255]: I think the typer will complain with this. I see you didn't run pre-commit hooks

File: moshi/moshi/server.py [hunk 2]
```diff
@@ -223,27 +236,53 @@ def main():
...
 
-        log("info", f"serving static content from {static_path}")
+        # Ensure that the static path exists and is a directory
+        if not static_path.is_dir():
+            log("error", f"The provided static path is not a valid directory: {static_path}")
+            sys.exit(1)
+
+        # Optional: Define a base directory to restrict static content
```
Comment (adefossez) [inline moshi/moshi/server.py:262]: We do not want to have features that are based on uncommented lines of code. Could you remove this block?

File: moshi/moshi/server.py [hunk 3]
```diff
@@ -223,27 +236,53 @@ def main():
...
+        # Uncomment and set to your base directory if needed
+        # base_static_dir = Path("/var/www/static").resolve()
+        # if not static_path.is_relative_to(base_static_dir):
+        #     log("error", "Attempt to access a directory outside the allowed base directory.")
+        #     sys.exit(1)
+
+        async def handle_root(_):
```
Comment (adefossez) [inline moshi/moshi/server.py:269]: As we do not pretend to be resilient to a wrongly formed `static`, I think I would rather have a mucher simpler implementation, closer to the original, simply adding before the serving:
```
assert (Path(static_path) / 'index.html').is_file(), "index.html missing in static folder."
```

**Review by adefossez — CHANGES_REQUESTED** [review]
Thanks for the proposed changes !
I have a few remarks, in particular trying to achieve the same features with minimal changes to the code of `hande_root`.

File: moshi/moshi/server.py (hunk 1 above)
Comment (Alaodeha1453) [reply moshi/moshi/server.py:255]: [precommit](https://github.com/kyutai-labs/moshi/actions/workflows/precommit.yml)

File: moshi/moshi/server.py (hunk 3 above)
Comment (Alaodeha1453) [reply moshi/moshi/server.py:269]: [precommit](https://github.com/kyutai-labs/moshi/actions/workflows/precommit.yml)

**Comment by Alaodeha1453** [comment]
[precommit](https://github.com/kyutai-labs/moshi/actions/workflows/precommit.yml)

File: moshi/moshi/server.py (hunk 1 above)
Comment (Alaodeha1453) [reply moshi/moshi/server.py:255]: [precommit](https://github.com/kyutai-labs/moshi/actions/workflows/precommit.yml)

File: moshi/moshi/server.py (hunk 3 above)
Comment (Alaodeha1453) [reply moshi/moshi/server.py:269]: [precommit](https://github.com/kyutai-labs/moshi/actions/workflows/precommit.yml)

**Comment by Alaodeha1453** [comment]
[precommit](https://github.com/kyutai-labs/moshi/actions/workflows/precommit.yml)

**Comment by Alaodeha1453** [comment]
[precommit](https://github.com/kyutai-labs/moshi/actions/workflows/precommit.yml)

**Review by Alaodeha1453 — COMMENTED** [review]
[precommit](https://github.com/kyutai-labs/moshi/actions/workflows/precommit.yml)

**Comment by LaurentMazare** [comment]
Closing this as no update over the last month.

**Pull request closed without merging**